add-constraint.md

```diff
@@ -51,11 +51,66 @@ Adding the [Check constraint](check.html) requires that all of a column's values
-```
+To add the [Foreign Key](foreign-key.html) constraint to an existing column or group of columns, you must first use [`CREATE INDEX`](create-index.html) to index the column or columns that you plan to constrain. Then you can use the `ADD CONSTRAINT` statement to add the Foreign Key constraint to the column or columns.
+
+For example, let's say you have two simple tables, `orders` and `customers`:
+
+~~~ sql
+> SHOW CREATE TABLE customers;
+~~~
+
+~~~
++-----------+-------------------------------------------------+
+|   Table   |                   CreateTable                   |
++-----------+-------------------------------------------------+
+| customers | CREATE TABLE customers (␤                       |
+|           |     id INT NOT NULL,␤                           |
+|           |     "name" STRING NOT NULL,␤                    |
+|           |     address STRING NULL,␤                       |
+|           |     CONSTRAINT "primary" PRIMARY KEY (id ASC),␤ |
+|           |     FAMILY "primary" (id, "name", address)␤     |
+|           | )                                               |
++-----------+-------------------------------------------------+
+(1 row)
+~~~
+
+~~~ sql
+> SHOW CREATE TABLE orders;
+~~~
+
+~~~
++--------+-------------------------------------------------------------------------------------------------------------+
+| Table  |                                                 CreateTable                                                 |
++--------+-------------------------------------------------------------------------------------------------------------+
+| orders | CREATE TABLE orders (␤                                                                                      |
+|        |     id INT NOT NULL,␤                                                                                       |
+|        |     customer_id INT NULL,␤                                                                                  |
+|        |     status STRING NOT NULL,␤                                                                                |
+|        |     CONSTRAINT "primary" PRIMARY KEY (id ASC),␤                                                             |
+|        |     FAMILY "primary" (id, customer_id, status),␤                                                            |
+|        |     CONSTRAINT check_status CHECK (status IN ('open':::STRING, 'complete':::STRING, 'cancelled':::STRING))␤ |
+|        | )                                                                                                           |
++--------+-------------------------------------------------------------------------------------------------------------+
+(1 row)
+~~~
+
+To ensure that each value in the `orders.customer_id` column matches a unique value in the `orders.id` column, you want to add the Foreign Key constraint to `orders.customer_id`. So you first create an index on `orders.customer_id`:
```

Comment:
yeah, probably doesn't need to be in the example, and might fit better on the FK page.

might be something like "an index on the columns in a foreign key is required because updates and deletes on the referenced table will need to search the referencing table for any matching records, to ensure those operations would not violate existing references. In practice, such index is likely also needed by the application using foreign keys, since finding all records which belong to some entity, for example all `orders` for a given `customer`, is very common."